src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,622 @@
+        3. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        4. Change the GPO permissions so the computer object is the only object that can read the policy.
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1' with ssh
+        2. Authenticate as 'deny_user1' with ssh
+    :expectedresults:
+        1. 'user1' authentication is successful
+        1. 'deny_user1' authentication is unsuccessful
+    :customerscenario: True
+    """
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(bz=1855281)
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_gpo__skips_unreadable_gpo_policies(client: Client, ad: AD, sssd_service_user: str):
+    """
+    :title: Group policy object host based access control skips unreadable GPO policies.
+    :description:
+        When a policy cannot be read, the default behavior is to block everything. To air on the
+        side of caution. Enabling 'ad_gpo_ignore_unreadable' should skip the unreadable policy
+        and process the policies that are readable.
+    :setup:
+        1. Create the following users; 'user1', 'user2'
+        2. Create and link the GPO 'test policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key. Add 'user2 to SeDenyInteractiveLogonRight key.
+        3. Create and link the GPO 'domain policy' and add user2' and group 'Domain Admins' to
+           SeInteractiveLogonRight key. Add 'user1' to SeDenyInteractiveLogonRight key and
+           set the policy to be enforcing.
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Modify the permissions to the GPO 'domain policy' and make it unreadable
+        6. Start SSSD
+    :steps:
+        1. Authenticate as 'user1' and 'user2'
+        2. Configure sssd.conf with 'ad_gpo_ignore_unreadable' = 'True' and restart SSSD
+        3. Authenticate as 'user1'
+        4. Authenticate as 'user2'
+    :expectedresults:
+        1. 'user1' and 'user2' authentication is unsuccessful
+        2. Configuration is updated with 'ad_gpo_ignore_unreadable' = 'True'
+        3. 'user1' authentication is successful
+        4. 'user2' authentication is unsuccessful
+    :customerscenario: True
+    """
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(bz=2151450)
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_gpo__works_when_auto_private_groups_is_set_true(client: Client, ad: AD, sssd_service_user: str):
```

Comment:
@thalman is there a functional way to test / simulate this bug? IIRC, we just ensured the patch contained an extra entry in the log when processing the group policy, and compared that output with the previous version.